tests/unittests/OperatorTest.cpp

```diff
@@ -580,6 +580,117 @@ TEST_P(InterpAndCPU, batchedReduceMeanQuantizedWithAxis) {
   }
 }
 
+/// Verify that batchedReduceMean optimization using AvgPool works correctly.
+TEST_P(InterpAndCPU, batchedReduceMeanUsingAvgPool) {
+  std::vector<size_t> dims = {3, 20, 4, 8};
+
+  size_t dimResMul = dims[0] * dims[1];
+  size_t dimTmpMul = dimResMul * dims[2];
+  size_t dimAllMul = dimTmpMul * dims[3];
+
+  auto *batch = mod_.createPlaceholder(ElemKind::FloatTy, dims, "batch", false);
+
+  auto IH = ctx_.allocate(batch)->getHandle();
+  IH.randomize(1.0, 100.0, mod_.getPRNG());
+
+  auto *R = F_->createBatchedReduceMean("reduce.mean", batch, {2, 3}, true);
+
+  auto *save = F_->createSave("save", R);
+  auto *result = ctx_.allocate(save->getPlaceholder());
+  EE_.compile(CompilationMode::Infer, F_);
+
+  EE_.run(ctx_);
+  auto H = result->getHandle();
+
+  // 1st reduce has i*j*k elements for a 4D (i,j,k,l) input & l is reduced.
+  std::vector<float> tmpRes(sizeof(float) * dimAllMul);
```

Comment:
Seems that I didn't cleanup & improve this test after getting it to work. Was giving less importance to test organization/effectiveness, and will fix that...

Reply:
@jfix71, ok, my initial idea was to introduce randomized dimensions sizes, which would not fit array's const initializers requirements (and I don't believe we would want to allocate memory on heap here ?). While I don't have randomization included in this PR, this would allow adding it by changing `dims() `variable initialization only.  Is this a good enough explanation or you still want this simplified ?

Reply:
I'd rather it be simplified down to a 4D float array, thanks @vuzelac-cadence.

Reply:
Done